hs_app_timeseries/receivers.py

```diff
@@ -62,25 +94,30 @@ def post_add_files_to_resource_handler(sender, **kwargs):
         if fl_ext == '.sqlite':
             validate_err_message = _validate_odm2_db_file(fl_obj_name)
             if not validate_err_message:
-                if extract_metadata:
-                    # first delete relevant metadata elements
+                # first delete relevant metadata elements
+                _delete_extracted_metadata(resource)
+                extract_err_message = _extract_metadata(resource, fl_obj_name)
+                if extract_err_message:
+                    # delete the invalid file
+                    delete_resource_file_only(resource, res_file)
+                    # cleanup any extracted metadata
                     _delete_extracted_metadata(resource)
-                    extract_err_message = _extract_metadata(resource, fl_obj_name)
+                    validate_files_dict['are_files_valid'] = False
+                    extract_err_message += " " + FILE_UPLOAD_ERROR_MESSAGE
+                    validate_files_dict['message'] = extract_err_message
```

Comment:
@pkdash it will be more performant if you put all error messages into a list, then use ```join``` in the end to get an aggregate error message since concatenation ```+``` is expensive.